packages/core/src/controllers/echo.ts

```diff
@@ -0,0 +1,30 @@
+import { generateChildLogger, Logger } from "@walletconnect/logger";
+import { IEcho } from "@walletconnect/types";
+import { ECHO_CONTEXT, ECHO_URL } from "../constants";
+
+export class Echo extends IEcho {
+  public readonly context = ECHO_CONTEXT;
+  constructor(public projectId: string, public logger: Logger) {
+    super(projectId, logger);
+    this.logger = generateChildLogger(logger, this.context);
+  }
+
+  public registerDeviceToken: IEcho["registerDeviceToken"] = async (params) => {
+    const { clientId, token, notificationType, enableEncrypted = false } = params;
+
+    const echoUrl = `${ECHO_URL}/${this.projectId}/clients`;
+
+    await fetch(echoUrl, {
```

Comment:
Don't we need to use `isomorphic-unfetch` or something here? Or is it already polyfilled